python/tests/test_ld_stats.py

```diff
@@ -0,0 +1,475 @@
+# MIT License
+#
+# Copyright (c) 2018-2019 Tskit Developers
+# Copyright (C) 2016 University of Oxford
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+Test cases for ld matrix calculations.
+"""
+import unittest
+import io
+
+import numpy as np
+import msprime
+
+import tskit
+import _tskit
+#import tests.tsutil as tsutil
+#import tests.test_wright_fisher as wf
+
+
+##############################
+# LD matrix calculator
+##############################
+
+
+## notes 6/12:
+## Question: Do we want to return the LD matrix over variants or over sites?
+##  I think sites, and that's what the naive_ld_matrix returns
```

Comment:
They're the same thing (well, in 1-to-1 correspondence, anyhow).

Reply:
I was thinking of the case of multiple variants at a single site. But I do think we want to return LD over all variants at a site

Reply:
Ah - more than two alleles? Sorry - thought you meant the objects returned by `ts.variants()`.  But yes - I agree.

Reply:
Right, more than two alleles.. I was using "allele" interchangeably with "variant", but I see now that those are different in terms of what `ts.variants()` returns. Sorry for the confusion